### PR TITLE
Added python script to setup.sh to automatically add all unit tests t…

### PR DESCRIPTION
**Pull request by MarkAHarley**
…o cmake and link IIR library to each test

### CODE REVIEW

File: unit_tests/cmake_linker.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        for test in testfiles:
+            cmake.write(f"\nadd_executable({test[:-3]} {test})\n")
+            cmake.write(f"target_link_libraries({test[:-3]} GTest::gtest_main)\n")
+            cmake.write(f"target_link_libraries({test[:-3]} iir::iir)\n")
+            cmake.write(f"gtest_discover_tests({test[:-3]})\n\n")
+
+
+
+if __name__ == "__main__":
+    main()
```
Comment (MHCooke) [inline unit_tests/cmake_linker.py:36]: Why have you mixed global variables with a main function. Imo stick to one or other

**Comment by MHCooke** [comment]
Would be good to add a dependency on python 3 to the readme

**Comment by MarkAHarley** [comment]
> Would be good to add a dependency on python 3 to the readme

Already done

File: unit_tests/cmake_linker.py (hunk 1 above)
Comment (MarkAHarley) [reply unit_tests/cmake_linker.py:36]: didn't think it was an issue since the globals are just constants

Comment (MHCooke) [reply unit_tests/cmake_linker.py:36]: This is python though, not C++, and we are supposed to be doing good software engineer practices. No such thing a constant in python

**Comment by MHCooke** [comment]
> > Would be good to add a dependency on python 3 to the readme
> 
> Already done

I don't see it in this PR, or on main?

File: unit_tests/cmake_linker.py (hunk 1 above)
Comment (MarkAHarley) [reply unit_tests/cmake_linker.py:36]: So are you proposing to just define them in the main function?

**Comment by MarkAHarley** [comment]
> > > Would be good to add a dependency on python 3 to the readme
> > 
> > 
> > Already done
> 
> I don't see it in this PR, or on main?

I did it on the update README branch because I assumed that's where we were keeping all that since it's still open. Could just do it on this branch though i guess.

**Review by MHCooke — APPROVED** [review]
Looks great!

**Pull request merged**